src/Voron/Data/Fixed/FixedSizeIterators.cs

```diff
@@ -164,6 +164,9 @@ public bool Skip(int count)
                 if (count != 0)
                     _pos += count;
 
+                if (_pos < 0)
+                    return false;
```

Comment:
I have run `CanSkipBeyondCountForEmbeddedIterator` test with the debugger and noticed that we got `-9` here. I was surprised but we allow to call `Skip()` with negative parameter (in our case skip was `10` so we passed `-10` and `_pos` was `1` that resulted in `-9`):

https://github.com/ravendb/ravendb/blob/a835aae9dd925ff77cac4e43da2436a388fae015/src/Voron/Data/Tables/Table.cs#L1331

I wonder if we shouldn't also make sure that `_pos` is valid (and I think that the minimal, valid value is `-1`). Now it's left with `-9` and _potentially_ someone could call `MoveNext()` and get AVE.

I think that `Skip(-10)` effectively should be the same as:

```
for(var i = 0; i<10;i++)
{
  if (it.MovePrev() == false)
     return false;
}

return true;
```

in that case `_pos` would be `-1` after returning false

Reply:
The same applies to passing very big parameter to `Skip()` - the max value of `_pos` should be `NumberOfEntries` I think

Reply:
Or maybe `Skip()` shouldn't modify `_pos` at all if it returns `false`?

Reply:
The behavior should be consistent with Embedded and LargeIterator

Reply:
This is wrong. If we pass `-skip`, we need to set it to `0`

Basically, we want to go back up to X steps.

It is equivalent to `last 10` and if I have less than 10, I want however many I have, from the first one.

Reply:
So to conclude, what do we want to do here? do we want to go over all those places that touch the `_pos` directly and ensure it can't be negative?

Right now it will _always_ cause AVE for such scenario, because getting a negative `_pos` will return `true` and hitting `MoveNext` because of it, will cause the AVE.

Reply:
Looking at the API usage, it looks like this is okay. Go ahead and change semantics in this manner, no need to clamp it.